app/services/location_lookup_service.rb

```diff
@@ -0,0 +1,33 @@
+class LocationLookupService
+  attr_reader :postcode
+
+  def initialize(postcode)
+    @postcode = postcode
+  end
+
+  def data
+    location_data = areas.map do |_, area|
+      next if area["codes"]["gss"].blank?
+
+      MapitPostcodeResponse.new(area)
+    end
+
+    location_data.compact
+  end
+
+private
+
+  def areas
+    return [] if response.blank?
+
+    response["response"].select { |data| data.first == "areas" }.first.last
+  end
+
+  def response
+    @response ||= begin
+                    JSON.parse(GdsApi.mapit.location_for_postcode(postcode).to_json)
+                  rescue GdsApi::HTTPNotFound
```

Comment:
just thinking we should also rescue the client error (same as they do here) https://github.com/alphagov/frontend/blob/8dbe0f0f6ca0f5a777a0d6ca77b858fe5adc2494/app/controllers/licence_controller.rb#L89?

Reply:
This is a good point. We're catching the 404 response so that user's will see the National Restrictions content as a default if Mapit can't find their postcode. Perhaps we should also specifically catch the BaseError so that users always see the National Restrictions content?

Reply:
ooh its hopefully fine. Lets put our faith in the API :) We can reassess later if needed!

Reply:
There are so many exceptions to choose from: https://github.com/alphagov/gds-api-adapters/blob/master/lib/gds_api/exceptions.rb